.github/workflows/build-and-push.yml

```
@@ -3,8 +3,10 @@ name: Create and publish a Docker image
 on:
   workflow_dispatch:  # allow to run manually
   push:
-   tags:
-    - '**'
+    branches:
+      - master
```

```suggestion
      - '**'
```
I would do the build step on all branches, just like we currently do with buildkite, but I would avoid tagging the release with `head` or `latest` outside the `master` branch.